updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -112,13 +112,31 @@ def add_handled_dependencies(dependency_names)
       @handled_dependencies[@current_directory] = set
     end
 
+    sig { params(dependencies: T::Array[{ name: T.nilable(String), directory: T.nilable(String) }]).void }
+    def add_handled_group_dependencies(dependencies)
+      raise "Current directory not set" if @current_directory == ""
```

Comment:
🔎 We're not using `@current_directory` in this method. Do we need this guard clause?

Reply:
I'm not sure if we need it -

`@current_directory` is referenced by the `DirectorySnapshot#dependency_files` and `DependencySnapshot#dependencies` methods, but I'm not sure how those are used.

If we remove the guard from here then it won't be called when `:dependency_has_directory` is enabled, so I'm erring on the side of leaving it in.